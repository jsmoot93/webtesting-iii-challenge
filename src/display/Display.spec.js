import React from "react";
import { render, fireEvent, cleanup } from "react-testing-library";
import "jest-dom/extend-expect";

import Dashboard from "../dashboard/Dashboard";
import Display from "./Display";

afterEach(cleanup);

describe("Display Tests", () => {
  it("Should match snapshot", () => {
    const { container } = render(<Display />);
    expect(container.firstChild).toMatchSnapshot();
  });
  it("Should display proper colors and text for each of the 3 states", () => {
    const { getByTestId } = render(<Dashboard />);
    const lockButton = getByTestId("lock-button");
    const openButton = getByTestId("open-button");
    const lockedLed = getByTestId("locked-led");
    const closedLed = getByTestId("closed-led");

    //Initially green and Unlocked/Open
    expect(lockedLed).toHaveTextContent("Unlocked");
    expect(lockedLed).toHaveClass("green-led");
    expect(closedLed).toHaveTextContent("Open");
    expect(closedLed).toHaveClass("green-led");

    //Closed gate should turn closed led red and text closed
    fireEvent.click(openButton);
    expect(lockedLed).toHaveTextContent("Unlocked");
    expect(lockedLed).toHaveClass("green-led");
    expect(closedLed).toHaveClass("red-led");
    expect(closedLed).toHaveTextContent("Closed");

    //Locking gate should change unlocked to red and text locked
    fireEvent.click(lockButton);
    expect(lockedLed).toHaveTextContent("Locked");
    expect(lockedLed).toHaveClass("red-led");
    expect(closedLed).toHaveClass("red-led");
    expect(closedLed).toHaveTextContent("Closed");
  });
});
