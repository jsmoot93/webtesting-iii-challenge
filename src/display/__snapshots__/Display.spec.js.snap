// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display Tests Should match snapshot 1`] = `
<div
  class="display panel"
>
  <div
    class="led green-led"
    data-testid="locked-led"
  >
    Unlocked
  </div>
  <div
    class="led green-led"
    data-testid="closed-led"
  >
    Open
  </div>
</div>
`;
